File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/ahn-product-1-0/Variable.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Variable : ISerialized
    {
        public Variable(XmlNode xmlNode)
        {
            XmlNode symbolNameNode = xmlNode.SelectSingleNode("symbolName");
            
            if (symbolNameNode != null)
            {
                if (symbolNameNode.Attributes["href"] != null || symbolNameNode.Attributes["id"] != null) 
                {
                    if (symbolNameNode.Attributes["id"] != null) 
                    {
                        symbolNameIDRef_ = symbolNameNode.Attributes["id"].Value;
                        XsdTypeToken ob = new XsdTypeToken(symbolNameNode);
                        IDManager.SetID(symbolNameIDRef_, ob);
                    }
                    else if (symbolNameNode.Attributes["href"] != null)
                    {
                        symbolNameIDRef_ = symbolNameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        symbolName_ = new XsdTypeToken(symbolNameNode);
                    }
                }
                else
                {
                    symbolName_ = new XsdTypeToken(symbolNameNode);
                }
            }
            
        
        }
        
    
        #region SymbolName_
        private XsdTypeToken symbolName_;
        public XsdTypeToken SymbolName_
        {
            get
            {
                if (this.symbolName_ != null)
                {
                    return this.symbolName_; 
                }
                else if (this.symbolNameIDRef_ != null)
                {
                    symbolName_ = IDManager.getID(symbolNameIDRef_) as XsdTypeToken;
                    return this.symbolName_; 
                }
                else
                {
                      return this.symbolName_; 
                }
            }
            set
            {
                if (this.symbolName_ != value)
                {
                    this.symbolName_ = value;
                }
            }
        }
        #endregion
        
        public string symbolNameIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/ahn-product-1-0/Variable.cpp
// Variable.cpp 
#include "Variable.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Variable::Variable(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //symbolNameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* symbolNameNode = xmlNode->FirstChildElement("symbolName");

   if(symbolNameNode){symbolNameIsNull_ = false;}
   else{symbolNameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- symbolNameNode , address : " << symbolNameNode << std::endl;
   #endif
   if(symbolNameNode)
   {
      if(symbolNameNode->Attribute("href") || symbolNameNode->Attribute("id"))
      {
          if(symbolNameNode->Attribute("id"))
          {
             symbolNameIDRef_ = symbolNameNode->Attribute("id");
             symbolName_ = boost::shared_ptr<XsdTypeToken>(new XsdTypeToken(symbolNameNode));
             symbolName_->setID(symbolNameIDRef_);
             IDManager::instance().setID(symbolNameIDRef_,symbolName_);
          }
          else if(symbolNameNode->Attribute("href")) { symbolNameIDRef_ = symbolNameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { symbolName_ = boost::shared_ptr<XsdTypeToken>(new XsdTypeToken(symbolNameNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeToken> Variable::getSymbolName()
{
   if(!this->symbolNameIsNull_){
        if(symbolNameIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeToken>(IDManager::instance().getID(symbolNameIDRef_));
        }else{
             return this->symbolName_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeToken>();
   }
}
}

